### PR TITLE
ActiveJob queue tracker

### PR DESCRIPTION
**Pull request by oneroyalace**
This PR moves all scraping tasks to `ActiveJob` and creates a status page to view the `ActiveJob`/sidekiq queue. The status page is updated using `ActionCable`, which pushes job queue changes to a websocket connection that the front end of Zenodotus subscribes to. 

Beginning with this PR, we'll need redis active to run Zenodotus in dev, as Sidekiq uses it for storage. 

To test:
1. `bundle install` (to install Sidekiq)
2. Run a local redis server
3. `bundle exec sidekiq`
4. `rails t`. Note: rails tests won't engage Sidekiq. To do that, run the following in the rails console:
```ruby
require 'sidekiq/api'
tweet_urls = [
  "https://twitter.com/JasonKander/status/1478016167679864836",
  "https://twitter.com/smwat/status/1478022399555739652",
  "https://twitter.com/greg_doucette/status/1477696042535735296",
  "https://twitter.com/BillAdairDuke/status/1478022935587614723",
  "https://twitter.com/seungminkim/status/1478022908538720261",
  "https://twitter.com/normative/status/1478021120251023361",
  "https://twitter.com/jeffjarvis/status/1478019870478548993",
  "https://twitter.com/thehill/status/1477678824280182790",
  "https://twitter.com/tressiemcphd/status/1476978397679075336",
  "https://twitter.com/RepLizCheney/status/1477723817200041988",
  "https://twitter.com/tressiemcphd/status/1478009292926201856",
  "https://twitter.com/ShaneGoldmacher/status/1477997873010819078",
  "https://twitter.com/farhip/status/1478000944042414082",
  "https://twitter.com/JonathanCohn/status/1477987859005116420",
  "https://twitter.com/PunchbowlNews/status/1478006865296601091",
  "https://twitter.com/NadiehBremer/status/1478003463430811650",
  "https://twitter.com/Martina/status/1477741805802246149",
  "https://twitter.com/TheEconomist/status/1478000715436072960",
  "https://twitter.com/brianstelter/status/1478000387185692673",
  "https://twitter.com/tcwittes/status/1477998719207456770",
  "https://twitter.com/frostyhk/status/1477872373072535552",
  "https://twitter.com/shmcminn/status/1477994333802541061",
  "https://twitter.com/ddale8/status/1477993680271253507",
  "https://twitter.com/NPRKelly/status/1477993406722985984",
  ]

tweet_urls.each do |url|
	Sources::Tweet.create_from_url! url
	puts "queue size: #{Sidekiq::Queue.new.size}"
end
```

You should see logs of the queue size, showing that the scraper tasks are being handled by SideKiq. Additionally, if you watch http://localhost:3000/jobs, you should see ActionCable updating the jobs tracker table there.  
To do list before merging this PR:
- [ ] Figure out how to manage Sidekiq workers' simultaneous requests to `Zorki`/`Forki`. Add more instances of the scrapers? Make them multithreaded?
- [ ]  Shrink the diff on this PR by merging #105 to master (which requires a little work on #105, namely adding some `erb` files)

### CODE REVIEW

**Comment by oneroyalace** [comment]
@cguess I've resolved merge conflicts on this branch, but we should probably merge #105 to master before this branch, as this branch is based off of that PR. 

Unfortunately there's not much in the way of testing on this branch because the changes are frontend-based (the status page) and queue based (move to sidekiq). 

To manually check that these changes are working though, you can run the following in the rails console:

```ruby
require 'sidekiq/api'
tweet_urls = [
  "https://twitter.com/JasonKander/status/1478016167679864836",
  "https://twitter.com/smwat/status/1478022399555739652",
  "https://twitter.com/greg_doucette/status/1477696042535735296",
  "https://twitter.com/BillAdairDuke/status/1478022935587614723",
  "https://twitter.com/seungminkim/status/1478022908538720261",
  "https://twitter.com/normative/status/1478021120251023361",
  "https://twitter.com/jeffjarvis/status/1478019870478548993",
  "https://twitter.com/thehill/status/1477678824280182790",
  "https://twitter.com/tressiemcphd/status/1476978397679075336",
  "https://twitter.com/RepLizCheney/status/1477723817200041988",
  "https://twitter.com/tressiemcphd/status/1478009292926201856",
  "https://twitter.com/ShaneGoldmacher/status/1477997873010819078",
  "https://twitter.com/farhip/status/1478000944042414082",
  "https://twitter.com/JonathanCohn/status/1477987859005116420",
  "https://twitter.com/PunchbowlNews/status/1478006865296601091",
  "https://twitter.com/NadiehBremer/status/1478003463430811650",
  "https://twitter.com/Martina/status/1477741805802246149",
  "https://twitter.com/TheEconomist/status/1478000715436072960",
  "https://twitter.com/brianstelter/status/1478000387185692673",
  "https://twitter.com/tcwittes/status/1477998719207456770",
  "https://twitter.com/frostyhk/status/1477872373072535552",
  "https://twitter.com/shmcminn/status/1477994333802541061",
  "https://twitter.com/ddale8/status/1477993680271253507",
  "https://twitter.com/NPRKelly/status/1477993406722985984",
  ]

tweet_urls.each do |url|
	Sources::Tweet.create_from_url! url
	puts "queue size: #{Sidekiq::Queue.new.size}"
end
```

You should see logs of the queue size, showing that the scraper tasks are being handled by SideKiq. Additionally, if you watch http://localhost:3000/jobs, you should see ActionCable updating the jobs tracker table there.

File: app/models/sources/instagram_post.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ class Sources::InstagramPost < ApplicationRecord
 
   # update materialized view when a new tweet is added
   after_commit on: [:create, :destroy] do
-    UnifiedTableRefreshJob.perform_later
+    # UnifiedTableRefreshJob.perform_later
```
Comment (cguess) [inline app/models/sources/instagram_post.rb:25]: Do we want to get rid of this?

File: app/models/sources/tweet.rb [hunk 2]
```diff
@@ -21,7 +21,7 @@ class Sources::Tweet < ApplicationRecord
 
   # update materialized view when a new tweet is added
   after_commit on: [:create, :destroy] do
-    UnifiedTableRefreshJob.perform_later
+    # UnifiedTableRefreshJob.perform_later
```
Comment (cguess) [inline app/models/sources/tweet.rb:24]: Is this unnecessary?

File: app/views/jobs_status/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  <table id="jobs-table" class="table-auto w-full">
+    <thead id="jobs-table-head">
+    <tr>
+      <th class="border text-center">Queue position</th>
+      <th class="border text-center">Enqueued at</th>
+      <th class="border text-center">Url to scrape</th>
+    </tr>
+    </thead>
+    <tbody id="jobs-table-body">
+      <%# unless @jobs_queue.empty? %>
```
Comment (cguess) [inline app/views/jobs_status/index.html.erb:11]: Remove this probably?

File: app/views/settings/index.html.erb [hunk 4]
```diff
@@ -32,28 +31,7 @@
     </tbody>
   </table>
 
+  <br>
```
Comment (cguess) [inline app/views/settings/index.html.erb:34]: These three `<br>` should probably be a CSS spacing instead

File: config/initializers/figaro.rb [hunk 5]
```diff
@@ -7,3 +7,7 @@
 # The URL and auth key for the external Zorki gem scraper
 Figaro.require_keys("ZORKI_SERVER_URL")
 Figaro.require_keys("ZORKI_AUTH_KEY")
+
+# The URL and auth key for the external Forki gem scraper
+# Figaro.require_keys("FORKI_SERVER_URL")
```
Comment (cguess) [inline config/initializers/figaro.rb:12]: Lets remove this if unneeded or uncomment it?

File: db/migrate/20211114190617_create_facebook_videos.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+class CreateFacebookVideos < ActiveRecord::Migration[6.1]
+  def change
+    create_table :facebook_videos, id: :uuid do |t|
+      t.references :facebook_post, type: :uuid, foreign_key: true
+      t.jsonb :video_data
+      t.timestamps
+    end
+  end
+end
+#             :user,
```
Comment (cguess) [inline db/migrate/20211114190617_create_facebook_videos.rb:10]: Let's remove these comments

File: test/channels/jobs_channel_test.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+require "test_helper"
+
+class JobsChannelTest < ActionCable::Channel::TestCase
+  # test "subscribes" do
```
Comment (cguess) [inline test/channels/jobs_channel_test.rb:4]: Should this be running?

File: test/controllers/jobs_tracker_controller_test.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class JobsTrackerControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
```
Comment (cguess) [inline test/controllers/jobs_tracker_controller_test.rb:4]: clear this out or put tests in it

**Review by cguess — CHANGES_REQUESTED** [review]

File: app/models/sources/instagram_post.rb (hunk 1 above)
Comment (oneroyalace) [reply app/models/sources/instagram_post.rb:25]: I think so. I'll make an issue for fixing text search

File: app/views/jobs_status/index.html.erb (hunk 3 above)
Comment (oneroyalace) [reply app/views/jobs_status/index.html.erb:11]: Yep

File: app/models/sources/tweet.rb (hunk 2 above)
Comment (oneroyalace) [reply app/models/sources/tweet.rb:24]: Yes, removed

File: app/views/settings/index.html.erb (hunk 4 above)
Comment (oneroyalace) [reply app/views/settings/index.html.erb:34]: This is changed in an incoming PR

**Review by cguess — APPROVED** [review]

**Pull request merged**